profiler/profile.go

```diff
@@ -44,48 +44,113 @@ const (
 	MetricsProfile
 )
 
-func (t ProfileType) String() string {
-	switch t {
-	case HeapProfile:
-		return "heap"
-	case CPUProfile:
-		return "cpu"
-	case MutexProfile:
-		return "mutex"
-	case BlockProfile:
-		return "block"
-	case GoroutineProfile:
-		return "goroutine"
-	case expGoroutineWaitProfile:
-		return "goroutinewait"
-	case MetricsProfile:
-		return "metrics"
-	default:
-		return "unknown"
+type profileType struct {
+	// Name is used to lookup the profile for pprof.Lookup(name) and returned by
+	// ProfileType.String(). For profile types that don't use this approach, the
+	// name can be chosen arbitrarily.
+	Name string
+	// Filename is the filename used for uploading the profile to the datadog
+	// backend.
+	// TODO(fg) are those filenames special / hardcoded in the backend?
+	Filename string
+	// TODO(fg) finalize API?
+	Collect func(profileType, *profiler) ([]byte, error)
```

Comment:
Then this could be `Bytes` or `Body`

Reply:
I think I still prefer `Collect()`. Reopen if you feel strongly about this.